File: src/intellibitz/sted/actions/LoadFontAction.java
package intellibitz.sted.actions;

import intellibitz.sted.ui.FontKeypad;
import intellibitz.sted.util.FileHelper;
import intellibitz.sted.util.Resources;

import javax.swing.*;
import java.awt.event.ActionEvent;
import java.io.File;

public class LoadFontAction
        extends STEDWindowAction {
    private FontKeypad fontKeypad;

    public LoadFontAction(FontKeypad fontSelectPanel) {
        super();
        putValue(Action.NAME, Resources.getSetting(Resources.LABEL_FONT_LOAD));
        fontKeypad = fontSelectPanel;
    }

    /**
     * Invoked when an action occurs.
     */
    public void actionPerformed(ActionEvent e) {
        loadFont();
    }

    private void loadFont() {
        final File file = FileHelper.openFont(getSTEDWindow());
        if (file != null) {
            fontKeypad.loadFont(file);
        }
    }


}

File: src/main/intellibitz/sted/actions/LoadFontAction.kt
package sted.actions

import sted.io.FileHelper.openFont
import sted.io.Resources
import sted.io.Resources.getSetting
import sted.ui.FontKeypad
import java.awt.event.ActionEvent

class LoadFontAction(fontSelectPanel: FontKeypad) : STEDWindowAction() {
    private val fontKeypad: FontKeypad

    /**
     * Invoked when an action occurs.
     */
    override fun actionPerformed(e: ActionEvent) {
        loadFont()
    }

    private fun loadFont() {
        val file = openFont(stedWindow)
        if (file != null) {
            fontKeypad.loadFont(file)
        }
    }

    init {
        putValue(NAME, getSetting(Resources.LABEL_FONT_LOAD))
        fontKeypad = fontSelectPanel
    }
}